### PR TITLE
Matching-Net

### PR DESCRIPTION
**Pull request by faezehamjadi**

### CODE REVIEW

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  venue:   "--"
+---
+
+### Description
+
+This paper proposed a solution for small data sets which suffers from overfitting and generalizing. The authors employ a metric learning which help the network to learn rapid from small data set. They applied this idea on a small set of Image Net and they get a accuracy from 87.6% to 93.2%.
+
+
+They use the advantages of one shot learning, but for prediction on the target, they investigate a metric to found how much a unseen sample is simmilar to seen sample in a episod and this similarity can present as a weight on classes.
+This metric plays such as attention or a kernel function. The authors used a cosin distance on two vectors as a metric.
```
[the remark this answers is not in the document]
Comment (jodoin) [reply deep-learning/_posts/2018-01-22-MatchingNet.md]: cosine

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Suppose $$(x_{i},y_{i})$$ is a given (input, class) and $$\bar{x}$$ is given input and predicted label $$\bar{y}$$ calculated as:
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-22-MatchingNet.md]: is a given input with a predicted label

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+---
+layout: review
+title:  "Matching Networks for One Shot Learning"
+tags:   "Deep Learning"
+author: Faezeh Amjadi
+pdf:    https://arxiv.org/pdf/1606.04080.pdf
+cite:
+  authors: "Oriol Vinyals, Charles Blundell, Timothy Lillicrap, Koray Kavukcuoglu, Daan Wierstra"
+  title:   "Matching Networks for One Shot Learning"
+  venue:   "--"
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-22-MatchingNet.md]: give the full Arxiv reference.

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+---
+
+### Description
+
+This paper proposed a solution for small data sets which suffers from overfitting and generalizing. The authors employ a metric learning which help the network to learn rapid from small data set. They applied this idea on a small set of Image Net and they get a accuracy from 87.6% to 93.2%.
```
[the remark this answers is not in the document]
Comment (jodoin) [reply deep-learning/_posts/2018-01-22-MatchingNet.md]: Do not understand "suffers from [...]generalizing?".  Maybe just "suffers from overfitting." is enough.

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+This metric plays such as attention or a kernel function. The authors used a cosin distance on two vectors as a metric.
+Suppose $$(x_{i},y_{i})$$ is a given (input, class) and $$\bar{x}$$ is given input and predicted label $$\bar{y}$$ calculated as:
+
+
+$$\bar{y} = \sum_{i=1}^k a(\bar{x},x_i) y_{i} $$
+ 
+
+
+a small support set contains of $$k$$ labell example with input $$x$$ and lable $$y$$, $$ S= {(x_{i},y_{i})}_{i=1}^k$$, given a new example $$\bar{x}$$, we want to know the probability that it is an example of a given class $$ P(\bar{y}\mid\bar{x},S)$$ where $$P$$ is parametrized by a neural network. Therefore $$S$$ maps to a classifier $$ c_{s}(\bar{x})$$. Simply, suppose a task $$T$$ has a distribution over lable sets $$L$$. First a set $$L$$ from task $$T$$ is sampled, Then from these sample labels, the support set $$S$$ and a batch $$B$$ are sampled. The Maching network has to minimize the prediction of lables in batch $$B$$ conditioned on support set $$(S)$$.
```
[the remark this answers is not in the document]
Comment (jodoin) [reply deep-learning/_posts/2018-01-22-MatchingNet.md]: The sentences "Simply, suppose a task $$T$$ has a distribution over lable sets $$L$$. First a set $$L$$ from task $$T$$ is sampled"  do not look correct.

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+<img src="/deep-learning/images/MNet/1.png" width="500">
+
+
+For a similarity metric, two embeding functions $$f$$ and $$g$$ need to take similarity on feature space X. function $$g$$ has embeded $$x_{i}$$ independently from other elements but $$S$$ could be able to effect how be embeded $$\bar{x}$$ through function $$f$$. Then an attention kernel calculate cosin distance on these functions (similar to nearest neighbor).
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-22-MatchingNet.md]: Do not forget to always start a sentence with a capital letter.  Here "Function $$g$$..."

Comment (jodoin) [reply deep-learning/_posts/2018-01-22-MatchingNet.md]: Not sure I understand "function $$g$$ has embeded $$x_{i}$$ independently from other elements but $$S$$ could be able to effect how be embeded $$\bar{x}$$ through function $$f$$."  Please reformulate.

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+<img src="/deep-learning/images/MNet/2.png" width="600">
+
+
+ 
+The embeding of $$x_{i}$$ is a neural network (VGG) follow by a Bi-LSTM. The function f is a neural network (VGG) and then the embeding function g applied to each element $$ x_{i}$$ to process the kernel for each set of S.
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-22-MatchingNet.md]: The embeding of x_{i}x is a neural network => do you mean that function g is a VGG net?

File: deep-learning/_posts/2018-01-22-MatchingNet.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+
+For a similarity metric, two embeding functions $$f$$ and $$g$$ need to take similarity on feature space X. function $$g$$ has embeded $$x_{i}$$ independently from other elements but $$S$$ could be able to effect how be embeded $$\bar{x}$$ through function $$f$$. Then an attention kernel calculate cosin distance on these functions (similar to nearest neighbor).
+
+<img src="/deep-learning/images/MNet/2.png" width="600">
+
+
+ 
+The embeding of $$x_{i}$$ is a neural network (VGG) follow by a Bi-LSTM. The function f is a neural network (VGG) and then the embeding function g applied to each element $$ x_{i}$$ to process the kernel for each set of S.  
+Note that $$\bar{y}$$ is a linear combination of the lables in support set.
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-22-MatchingNet.md]: in the support set S.

**Comment by ppoulin91** [comment]
I have 2 meta-comments :
- We might need to review your git setup, you commit using a different "identity" as your git account
- Try not to commit using the same commit message twice, or otherwise amend your commit to delete the previous one

File: deep-learning/_posts/2018-01-22-MatchingNet.md (hunk 4 above)
Comment (ppoulin91) [inline deep-learning/_posts/2018-01-22-MatchingNet.md]: suffer

**Comment by ppoulin91** [comment]
You should also try to split your paragraphs on multiple lines (one sentence / line), as it makes reviewing easier.

**Comment by ppoulin91** [comment]
I think I might try to review after the first batch of corrections
